### DISCUSSION TITLE
Do we need the lg-heading component?

### DISCUSSION THREAD

**Discussion by owensgit** [Legal-and-General/canopy#468] [General] [answered]
Since we removed the class input on the `lg-heading` component, I've noticed there's been many times where I've wanted to modify the heading, such as making it a different size, or adding extra margin because the component currently doesn't support the margin directive (https://github.com/Legal-and-General/canopy/issues/171). 

So I've started to just use the HTML heading tags directly, instead. e.g. `<h1 class="lg-font-size-1">Badgers are cool</h1>`.

This leads me to think, do we need the lg-heading component? What value does it add? I'd like to suggest that we either:

1. Remove the component entirely and recommend that adopters use the HTML tags directly, or...
2. Re-introduce the class input, so we can customise the heading where required

**Comment by paul-request** [comment]
Given the styles are global already and that component doesn't give us much more than a wrapper around HTML heading tags, I would opt for option 1. If we went for option 2, it opens it up to just as much abuse as just allowing people to use plain old HTML headings with their own style modifications, so seems like an unnecessary overhead to me.

**Reply by aaronBery** [reply]
An attribute directive could be another option e.g. `<h1 lgHeading>Snakes are scary</h1>` As it can apply the necessary classes automatically. There is also scope to adopt inputs if you want to allow some level of customisation

**Reply by Hopkinsa** [reply]
From a purely selfish point of view, I would personally prefer option 1 (`<h1 class="lg-font-size-1">Badgers are cool</h1>`), as in the design for Seal there are different sizes for `<h1>` and this approach allows the sizes to be set clearly, without the need of directives that may lead to confusion.

**Reply by aaronBery** [reply]
We've taken the directive approach with other elements before e.g. tables, so it's not something completely new although the table is obviously a more complex scenario where using techniques such directives has a very clear benefit.

That being said the directive approach for headings has some benefits. You can still customise using inputs as you would if you were setting classes. `<h1 lgHeading size="md">Slugs are gross</h1>` is an equivalent of something like `<h1 class="lg-font-size-md">Slugs are gross</h1>`.

It's worth stating the directive approach doesn't fully lock down the heading for abuse as users can still add any class they want however it steers them away from the practice of manually adding classes and towards using the component api.

**Reply by elenagarrone** [reply]
The main reason for the `lg-heading` component to exist is not for styling but for accessibility - to enforce the correct markup. 
In certain components we want to force a user to use the correct markup e.g. in the accordion the toggle should always be a heading. I've seen many cases where users were using a regular paragraph and that's not right. 
I do believe the `lg-heading` component is useful in these kind of scenarios. 

Is there a reason why we can't keep the component and use it within other Canopy components? 
In all the other scenarios you can use whatever you want.. it doesn't have to be enforced 🤔 

Agree that we should create directives for the font-sizes.

**Comment by owensgit** [comment, marked as answer]
After reading [Elena's comment](https://github.com/Legal-and-General/canopy/discussions/468#discussioncomment-1157752) and discussing with her, I think it's a good idea to keep the component like she mentioned. The `lg-heading` component allows you to pass a `level` which other Canopy components use to enforce the use of semantic headings. E.g. `lg-accordion-heading` also accepts a `level` and is using `lg-heading` under the hood. Removing `lg-heading` would mean repeating this logic across several other components. It's also used in this way in other apps, not just internally.

For this reason, I think we should keep  `lg-heading` as-is, and then normalise the use of HTML heading tags directly in most normal circumstances, e.g. `<h1 class="lg-font-size-1">Foxes are sneaky</h1>`.
